### ISSUE TITLE
GZIP body decompression before decoding through logging side effect

### ISSUE THREAD

**Issue by OlgaMaciaszek** [OpenFeign/feign#1399]
While upgrading Feign in SC OpenFeign to versions: 11.1, 11.2 (btw., I can't see this tag in repo, but the artifact is present in mvn), I have stumbled upon an issue:

When `GZIP` is set as `CONTENT-ENCODING`, [the response body is now being wrapped with `GZIPInputStream`](https://github.com/OpenFeign/feign/blob/4f2baf89c00dbad47eb51e0377820d6fb5feb71a/core/src/main/java/feign/Client.java#L133-L134). When logging is not enabled, a response this `GZIPInputStream` body is then passed to decoders for further handling.

However, when logging is enabled, the GZIP body is decompressed [here](https://github.com/OpenFeign/feign/blob/4f2baf89c00dbad47eb51e0377820d6fb5feb71a/core/src/main/java/feign/Logger.java#L99) (through calling `read(byte[] buf, int off, int len)` from `java.util.zip.GZIPInputStream.java`) and then this decompressed byte array is set as response body [here](https://github.com/OpenFeign/feign/blob/4f2baf89c00dbad47eb51e0377820d6fb5feb71a/core/src/main/java/feign/Logger.java#L105).

**Comment by velo** [comment]
> While upgrading Feign in SC OpenFeign to versions: 11.1, 11.2 (btw., I can't see this tag in repo, but the artifact is present in mvn)

This is odd I can see it fine: https://github.com/OpenFeign/feign/releases/tag/11.2

**Comment by OlgaMaciaszek** [comment]
Right, my bad: the tag is there; it's the release notes that are missing; sorry for the confusion.

**Comment by kdavisk6** [comment]
This side-effect has come up a few times.  I wonder if we need to reconsider this approach, or at least make it clear that logging will read the response and potentially re-wrap.
